### ISSUE TITLE
Sync DB schema loses manual schema config

### ISSUE THREAD

**Issue by notrom** [metabase/metabase#12680]
**Describe the bug**
After changing source DB schema (adding a column), the Metabase schema sync loses all the manual configuration to the table that has been changed.

**Logs**
```
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:17:41+12:00 INFO metabase.sync.util STARTING: step 'sync-fields' for sqlserver Database 2 'blah'
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:17:42+12:00 INFO metabase.sync.sync-metadata.fields.sync-metadata Special type of Table 156 'rel.MB_test' Field 'one' has changed from ':type/PK' to 'null'.
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:17:42+12:00 INFO metabase.sync.sync-metadata.fields.sync-metadata Special type of Table 156 'rel.MB_test' Field 'two' has changed from ':type/Category' to 'null'.
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:17:42+12:00 INFO metabase.sync.sync-metadata.fields.sync-metadata Special type of Table 156 'rel.MB_test' Field 'three' has changed from ':type/Category' to 'null'.
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:17:42+12:00 INFO metabase.sync.util FINISHED: step 'sync-fields' for sqlserver Database 2 'blah' (1.2 s)
... Manually set some config on that able again ...
... After removing the "four" column from the table again ...
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:28:01+12:00 INFO metabase.sync.util STARTING: step 'sync-fields' for sqlserver Database 2 'blah'
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:28:03+12:00 INFO metabase.sync.sync-metadata.fields.sync-instances Marking Field 'Table 156 'rel.MB_test' Field 'four'' as inactive.
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:28:03+12:00 INFO metabase.sync.sync-metadata.fields.sync-metadata Special type of Table 156 'rel.MB_test' Field 'one' has changed from ':type/PK' to 'null'.
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:28:03+12:00 INFO metabase.sync.sync-metadata.fields.sync-metadata Special type of Table 156 'rel.MB_test' Field 'two' has changed from ':type/Category' to 'null'.
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:28:03+12:00 INFO metabase.sync.sync-metadata.fields.sync-metadata Special type of Table 156 'rel.MB_test' Field 'three' has changed from ':type/Category' to 'null'.
[abdd05e7-21ee-41d1-b3c1-dc02b542137b] 2020-06-11T16:28:03+12:00 INFO metabase.sync.util FINISHED: step 'sync-fields' for sqlserver Database 2 'blah' (1.4 s)
```

**To Reproduce**
1. Create a new view/table in source database
```SQL
create view [rel].[MB_test]
as
Select 1 as one, 2 as two, 3 as three
GO
```
2. In Metabase do a schema sync. View is visible in the Metabase data model (after a browser refresh, another issue?) Change column "one" to be Entity Key
3. Update view/table
```SQL
alter view [rel].[MB_test]
as
Select 1 as one, 2 as two, 3 as three, 4 as four
GO
```
4. Sync Metabase schema again. In the Metabase data model the existing fields are all set to "No Special Type"

**Expected behavior**
Metabase should remember the manual (and previously determined automatic) config items in the schema. I'd understand if I'd changed column names etc. but that's not he case.

**Screenshots**
Before adding new column:
![image](https://user-images.githubusercontent.com/4504437/84345810-71f4c480-ac02-11ea-87d3-97d6ad08ffdb.png)

After adding new column:
![image](https://user-images.githubusercontent.com/4504437/84345828-82a53a80-ac02-11ea-93b7-eb2067a31cc9.png)

**Information about your Metabase Installation:**
```
{
  "browser-info": {
    "language": "en-US",
    "platform": "Win32",
    "userAgent": "Mozilla/5.0 (Windows NT 10.0; Win64; x64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/83.0.4103.61 Safari/537.36",
    "vendor": "Google Inc."
  },
  "system-info": {
    "file.encoding": "Cp1252",
    "java.runtime.name": "OpenJDK Runtime Environment",
    "java.runtime.version": "13.0.2+8",
    "java.vendor": "Oracle Corporation",
    "java.vendor.url": "https://java.oracle.com/",
    "java.version": "13.0.2",
    "java.vm.name": "OpenJDK 64-Bit Server VM",
    "java.vm.version": "13.0.2+8",
    "os.name": "Windows 10",
    "os.version": "10.0",
    "user.language": "en",
    "user.timezone": "Pacific/Auckland"
  },
  "metabase-info": {
    "databases": [
      "sqlserver",
      "postgres",
      "h2"
    ],
    "hosting-env": "unknown",
    "application-database": "postgres",
    "application-database-details": {
      "database": {
        "name": "PostgreSQL",
        "version": "12.1"
      },
      "jdbc-driver": {
        "name": "PostgreSQL JDBC Driver",
        "version": "42.2.8"
      }
    },
    "run-mode": "prod",
    "version": {
      "date": "2020-05-28",
      "tag": "v0.35.4",
      "branch": "release-0.35.x",
      "hash": "b3080fa"
    },
    "settings": {
      "report-timezone": "Pacific/Auckland"
    }
  }
}
```

**Severity**
Blocker, I can't change my data warehouse schema without losing Metabase schema settings for changed tables. If I lose those settings lots of questions simply don't work and my data model no longer functions, I don't have the time to re-apply these settings. Lots of the benefit we see from Metabase is based around the benefits of the data model.

**Additional context**
Also have the same issue when removing that "four" column again.

:arrow_down: Please click the :+1: reaction instead of leaving a `+1` or `update?` comment

**Comment by flamber** [comment]
Yeah, that makes it difficult to make schema changes.
So currently, the only workaround I see is to save the manual data of `metabase_field` (for the table being updated) and reapply after the schema changes. Not a very optimal workaround.
Related to #12574

**Comment by notrom** [comment]
@flamber this might catch people out and it's a quite a nasty surprise with quite a bit of unexpected rework. Any chance of bumping the priority?

**Comment by flamber** [comment]
@notrom I'm fairly sure it would depend on #12574 (which umbrellas a couple other issues). We need to dig deeper into this, but for now, I don't think it's a P1, since most people will likely not change their schema - and since this is how it's been working forever (from what I can tell) and only reported now - perhaps P1.5

**Comment by jlisyc** [comment]
Hi all. 
Do you think it's going to be solved in 0.35.5?

**Comment by Nico-P-FR** [comment]
Same problem for us. We update a lot our schemas and this bug generate a lot of rework the last weeks. Furthermore it lead to broken dashboard/cards when they used "foreign key automatic join" in filters / group by instead of a real join.

**Comment by jlisyc** [comment]
Hi @flamber , Do you know something about dates to solved this issue? Could be for 0.35.5? Thanks a lot.

**Comment by sbelak** [comment]
It's fixed in #12802 and will be included in the next release (either 0.35.5 or 0.36.0).

**Comment by jlisyc** [comment]
Thanks!
